sdk/core/core-rest-pipeline/src/policies/exponentialRetryPolicy.ts

```diff
@@ -59,117 +40,5 @@ export interface ExponentialRetryPolicyOptions {
-      statusCode === 505
-    ) {
-      return false;
-    }
-
-    const currentCount = retryData && retryData.retryCount;
-
-    return currentCount < retryCount;
-  }
-
-  /**
-   * Updates the retry data for the next attempt.
-   *
-   * @param retryData -  The retry data.
-   * @param err - The operation's error, if any.
-   */
-  function updateRetryData(retryData: RetryData, err?: RetryError): RetryData {
-    if (err) {
-      if (retryData.error) {
-        err.innerError = retryData.error;
-      }
-
-      retryData.error = err;
-    }
-
-    // Adjust retry count
-    retryData.retryCount++;
-
-    // Exponentially increase the delay each time
-    const exponentialDelay = retryInterval * Math.pow(2, retryData.retryCount);
-    // Don't let the delay exceed the maximum
-    const clampedExponentialDelay = Math.min(maxRetryInterval, exponentialDelay);
-    // Allow the final value to have some "jitter" (within 50% of the delay size) so
-    // that retries across multiple clients don't occur simultaneously.
-    const delayWithJitter =
-      clampedExponentialDelay / 2 + getRandomIntegerInclusive(0, clampedExponentialDelay / 2);
-
-    retryData.retryInterval = delayWithJitter;
-
-    return retryData;
-  }
-
-  async function retry(
-    next: SendRequest,
-    retryData: RetryData,
-    request: PipelineRequest,
-    response?: PipelineResponse,
-    requestError?: RetryError
-  ): Promise<PipelineResponse> {
-    retryData = updateRetryData(retryData, requestError);
-    const isAborted = request.abortSignal?.aborted;
-    if (!isAborted && shouldRetry(response, retryData)) {
-      logger.info(`Retrying request in ${retryData.retryInterval}`);
-      try {
-        await delay(retryData.retryInterval);
-        const res = await next(request);
-        return retry(next, retryData, request, res);
-      } catch (e) {
-        return retry(next, retryData, request, response, e);
-      }
-    } else if (isAborted || requestError || !response) {
-      // If the operation failed in the end, return all errors instead of just the last one
-      const err =
-        retryData.error ||
-        new RestError("Failed to send the request.", {
-          code: RestError.REQUEST_SEND_ERROR,
-          statusCode: response?.status,
-          request: response?.request,
-          response
-        });
-      throw err;
-    } else {
-      return response;
-    }
-  }
-
-  return {
-    name: exponentialRetryPolicyName,
-    async sendRequest(request: PipelineRequest, next: SendRequest): Promise<PipelineResponse> {
-      const retryData = {
-        retryCount: 0,
-        retryInterval: 0
-      };
-      try {
-        const response = await next(request);
-        return retry(next, retryData, request, response);
-      } catch (e) {
-        const error: RestError = e;
-        return retry(next, retryData, request, error.response, error);
-      }
-    }
-  };
+  return retryPolicy(exponentialRetryStrategy(options));
```

Comment:
This cleanup btw

![](https://i.pinimg.com/originals/c5/32/91/c5329178c33319e847fc75f39f43241f.gif)